src/documents/file_handling.py

```diff
@@ -128,13 +128,31 @@ def generate_unique_filename(doc, archive_filename=False):
 
 def generate_filename(doc, counter=0, append_gpg=True, archive_filename=False):
     path = ""
+    filename_format = settings.FILENAME_FORMAT
 
     try:
-        if settings.PAPERLESS_FILENAME_FORMAT is not None:
-            tags = defaultdictNoStr(lambda: slugify(None), many_to_dictionary(doc.tags))
+        if doc.storage_path:
+            if doc.storage_path.path is not None:
```

Comment:
Is this condition possible?  I believe the model doesn't allow a null for the field.